v4/export/dump.go

```diff
@@ -215,3 +254,31 @@ Loop:
 	}
 	return true, nil
 }
+
+func updateServiceSafePoint(ctx context.Context, pdClient pd.Client, ttl int64, snapshotTS uint64) {
+	updateInterval := time.Duration(ttl/2) * time.Second
+	tick := time.NewTicker(updateInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-tick.C:
+			log.Debug("update PD safePoint limit with ttl",
+				zap.Uint64("safePoint", snapshotTS),
+				zap.Int64("ttl", ttl))
+			for {
+				_, err := pdClient.UpdateServiceGCSafePoint(ctx, dumplingServiceSafePointID, ttl, snapshotTS)
+				if err == nil {
+					break
+				}
+				log.Warn("update PD safePoint failed", zap.Error(err))
+				select {
+				case <-ctx.Done():
+					return
+				case <-time.After(time.Second):
+				}
+			}
```

Comment:
Note that PD and Dumpling are not necessarily living in the same cluster. So failure to update the TTL should *not* leading to this never-ending warning.